{
  "errors": [],
  "generated_at": "2024-10-27T18:00:49Z",
  "metrics": {
    "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\failure_demo.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 24,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\pytest_timeout.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 456,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\setup.py": {
      "CONFIDENCE.HIGH": 0,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 0,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 4,
      "nosec": 0,
      "skipped_tests": 0
    },
    "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py": {
      "CONFIDENCE.HIGH": 23,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 23,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 525,
      "nosec": 0,
      "skipped_tests": 0
    },
    "_totals": {
      "CONFIDENCE.HIGH": 23,
      "CONFIDENCE.LOW": 0,
      "CONFIDENCE.MEDIUM": 0,
      "CONFIDENCE.UNDEFINED": 0,
      "SEVERITY.HIGH": 0,
      "SEVERITY.LOW": 23,
      "SEVERITY.MEDIUM": 0,
      "SEVERITY.UNDEFINED": 0,
      "loc": 1009,
      "nosec": 0,
      "skipped_tests": 0
    }
  },
  "results": [
    {
      "code": "67     )\n68     assert \"++ Timeout ++\" in result.stdout.lines[-1]\n69 \n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 68,
      "line_range": [
        68
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "95     )\n96     assert \"++ Timeout ++\" in result.stdout.lines[-1]\n97 \n",
      "col_offset": 4,
      "end_col_offset": 53,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 96,
      "line_range": [
        96
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "109     result = pytester.runpytest_subprocess()\n110     assert result.ret > 0\n111 \n",
      "col_offset": 4,
      "end_col_offset": 25,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 110,
      "line_range": [
        110
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "150     result = pytester.runpytest_subprocess(\"--timeout=1\", f\"--timeout-method={meth}\")\n151     assert result.ret > 0\n152     assert \"Timeout\" in result.stdout.str() + result.stderr.str()\n",
      "col_offset": 4,
      "end_col_offset": 25,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 151,
      "line_range": [
        151
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "151     assert result.ret > 0\n152     assert \"Timeout\" in result.stdout.str() + result.stderr.str()\n153 \n",
      "col_offset": 4,
      "end_col_offset": 65,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 152,
      "line_range": [
        152
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "171     result = pytester.runpytest_subprocess(\"--timeout=1\")\n172     assert result.ret == 0\n173     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n",
      "col_offset": 4,
      "end_col_offset": 26,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 172,
      "line_range": [
        172
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "172     assert result.ret == 0\n173     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n174 \n",
      "col_offset": 4,
      "end_col_offset": 69,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 173,
      "line_range": [
        173
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "199     )\n200     assert result.ret > 0\n201     assert \"Timeout\" in result.stdout.str() + result.stderr.str()\n",
      "col_offset": 4,
      "end_col_offset": 25,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 200,
      "line_range": [
        200
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "200     assert result.ret > 0\n201     assert \"Timeout\" in result.stdout.str() + result.stderr.str()\n202 \n",
      "col_offset": 4,
      "end_col_offset": 65,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 201,
      "line_range": [
        201
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "224     result = pytester.runpytest_subprocess(\"--timeout=1\", \"-s\")\n225     assert result.ret == 0\n226     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n",
      "col_offset": 4,
      "end_col_offset": 26,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 225,
      "line_range": [
        225
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "225     assert result.ret == 0\n226     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n227 \n",
      "col_offset": 4,
      "end_col_offset": 69,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 226,
      "line_range": [
        226
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "344     result = pytester.runpytest_subprocess()\n345     assert result.ret\n346 \n",
      "col_offset": 4,
      "end_col_offset": 21,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 345,
      "line_range": [
        345
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "367     result = pytester.runpytest_subprocess()\n368     assert result.ret == 0\n369 \n",
      "col_offset": 4,
      "end_col_offset": 26,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 368,
      "line_range": [
        368
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "391     result = pytester.runpytest_subprocess()\n392     assert result.ret == 0\n393 \n",
      "col_offset": 4,
      "end_col_offset": 26,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 392,
      "line_range": [
        392
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "411     result = pytester.runpytest_subprocess()\n412     assert \"=== 1 failed in \" not in result.outlines[-1]\n413 \n",
      "col_offset": 4,
      "end_col_offset": 56,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 412,
      "line_range": [
        412
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "431     result = pytester.runpytest_subprocess(\"--timeout=1\", \"-s\")\n432     assert result.ret == 0\n433     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n",
      "col_offset": 4,
      "end_col_offset": 26,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 432,
      "line_range": [
        432
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "432     assert result.ret == 0\n433     assert \"Timeout\" not in result.stdout.str() + result.stderr.str()\n434 \n",
      "col_offset": 4,
      "end_col_offset": 69,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 433,
      "line_range": [
        433
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "482         child.terminate(force=True)\n483     assert \"timeout >1.0s\" not in result\n484     assert \"fail\" not in result\n",
      "col_offset": 4,
      "end_col_offset": 40,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 483,
      "line_range": [
        483
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "483     assert \"timeout >1.0s\" not in result\n484     assert \"fail\" not in result\n485 \n",
      "col_offset": 4,
      "end_col_offset": 31,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 484,
      "line_range": [
        484
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "526         child.terminate(force=True)\n527     assert \"timeout >1.0s\" in result\n528     assert \"fail\" in result\n",
      "col_offset": 4,
      "end_col_offset": 36,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 527,
      "line_range": [
        527
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "527     assert \"timeout >1.0s\" in result\n528     assert \"fail\" in result\n529 \n",
      "col_offset": 4,
      "end_col_offset": 27,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 528,
      "line_range": [
        528
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "533 \n534     assert not pytest_timeout.is_debugging()\n535 \n",
      "col_offset": 4,
      "end_col_offset": 44,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 534,
      "line_range": [
        534
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    },
    {
      "code": "548 \n549     assert pytest_timeout.is_debugging(custom_trace)\n550 \n",
      "col_offset": 4,
      "end_col_offset": 52,
      "filename": "D:\\bandit\\upstream-repos\\py3-pytest-timeout\\test_pytest_timeout.py",
      "issue_confidence": "HIGH",
      "issue_cwe": {
        "id": 703,
        "link": "https://cwe.mitre.org/data/definitions/703.html"
      },
      "issue_severity": "LOW",
      "issue_text": "Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.",
      "line_number": 549,
      "line_range": [
        549
      ],
      "more_info": "https://bandit.readthedocs.io/en/1.7.10/plugins/b101_assert_used.html",
      "test_id": "B101",
      "test_name": "assert_used"
    }
  ]
}